nav2_docking/README.md

```diff
@@ -36,12 +36,14 @@ The `ChargingDock` and `NonChargingDock` plugins are the heart of the customizab
 The docking procedure is as follows:
 1. Take action request and obtain the dock's plugin and its pose
 2. If the robot is not within the prestaging tolerance of the dock's staging pose, navigate to the staging pose
-3. Use the dock's plugin to initially detect the dock and return the docking pose
-4. Enter a vision-control loop where the robot attempts to reach the docking pose while its actively being refined by the vision system
-5. Exit the vision-control loop once contact has been detected or charging has started
-6. Wait until charging starts (if applicable) and return success.
+3. Call the dock's plugin `startDetectionProcess()` method to activate any external detection mechanisms.
+4. Use the dock's plugin to initially detect the dock (`getRefinedPose`) and return the docking pose.
+5. Enter a vision-control loop where the robot attempts to reach the docking pose while it's actively being refined by the vision system.
+6. Exit the vision-control loop once contact has been detected or charging has started (if applicable).
+7. Wait until charging starts (if applicable) and return success.
+8. Call the dock's plugin `stopDetectionProcess()` method to deactivate any external detection mechanisms (this is also called on failure/cancellation).
 
-If anywhere this procedure is unsuccessful, `N` retries may be made by driving back to the dock's staging pose and trying again. If still unsuccessful, it will return a failure code to indicate what kind of failure occurred to the client.
+If anywhere this procedure is unsuccessful (before step 8), `N` retries may be made. This involves calling `stopDetectionProcess()`, driving back to the dock's staging pose, and then restarting the process from step 3. If still unsuccessful after retries, it will return a failure code to indicate what kind of failure occurred to the client.
```

Comment:
```suggestion
If anywhere this procedure is unsuccessful (before step 8), `N` retries may be made, driving back to the dock's staging pose, and then restarting the process from step 3. If still unsuccessful after retries, it will return a failure code to indicate what kind of failure occurred to the client.
```

Reply:
TMI